cirq-core/cirq/experiments/z_phase_calibration_test.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+import cirq
+
+from cirq.experiments.z_phase_calibration import (
+    calibrate_z_phases,
+    z_phase_calibration_workflow,
+    plot_z_phase_calibration_result,
+)
+from cirq.experiments.xeb_fitting import XEBPhasedFSimCharacterizationOptions
+
+_ANGLES = ['theta', 'phi', 'chi', 'zeta', 'gamma']
+
+
+def _create_tests(n, seed):
+    rng = np.random.default_rng(seed)
+    angles = (rng.random((n, 5)) * 2 - 1) * np.pi
+    # Add errors to the first 2 angles (theta and phi).
+    # The errors for theta and phi are in the union (-2, -1) U (1, 2).
+    # This is because we run the tests with few repetitions so a small error might not get fixed.
+    error = np.concatenate(
+        [(rng.random((n, 2)) + 1) * rng.choice([-1, 1], (n, 2)), np.zeros((n, 3))], axis=-1
+    )
+    return zip(angles, error)
+
+
+def _trace_distance(A, B):
+    return 0.5 * np.abs(np.linalg.eigvals(A - B)).sum()
+
+
+class _TestSimulator(cirq.Simulator):
+    """A simulator that replaces a specific gate by another."""
+
+    def __init__(self, gate: cirq.Gate, replacement: cirq.Gate, **kwargs):
+        super().__init__(**kwargs)
+        self.gate = gate
+        self.replacement = replacement
+
+    def _core_iterator(
+        self,
+        circuit: 'cirq.AbstractCircuit',
+        sim_state,
+        all_measurements_are_terminal: bool = False,
+    ):
+        new_circuit = cirq.Circuit(
+            [
+                [op if op.gate != self.gate else self.replacement(*op.qubits) for op in m]
+                for m in circuit
+            ]
+        )
+        yield from super()._core_iterator(new_circuit, sim_state, all_measurements_are_terminal)
+
+
+@pytest.mark.parametrize(['angles', 'error'], _create_tests(n=10, seed=32432432))
+def test_calibrate_z_phases(angles, error):
```

Comment:
The tests are very slow and take about 1 minute to run on my desktop with `--numprocesses=2` (number of cores on GHA).  They make up the bulk of the 20 slowest durations in this PRs CI at https://github.com/quantumlib/Cirq/actions/runs/11152830363/job/30999210792?pr=6728#step:7:507.

Please reduce the number and computational size of the tests so we don't add more than several seconds to the pytest time.  The main purpose of unit tests is to exercise all code branches and check a few small and quick computations.
If you need to run larger tests, please move them to a separate test method decorated with `@pytest.mark.slow`.

Reply:
they should be faster now, ptal